### ISSUE TITLE
Issue with Reminders

### ISSUE THREAD

**Issue by SruiSan** [glpi-project/glpi#10654]
### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

9.5.7

### Bug description

The public reminders don't show. You only can show your own reminders

This is the Web Interface

![SnapShot_GLPI_Web](https://user-images.githubusercontent.com/45992623/154048381-4cf38860-15f1-4bcc-84ea-e38e4ca128be.png)

Only see "Personal Reminders", but in database:

![SnapShot_GLPI_DB](https://user-images.githubusercontent.com/45992623/154048441-6e9d0478-9e8e-45bc-9f04-f0bbe87c8e06.png)

In 9.5.5 we can view all without issues.

I'm debug the query for Public Reminders:

```
SELECT DISTINCT `glpi_reminders`.*
FROM `glpi_reminders`
LEFT JOIN `glpi_reminders_users` ON (`glpi_reminders_users`.`reminders_id` = `glpi_reminders`.`id`)
LEFT JOIN `glpi_groups_reminders` ON (`glpi_groups_reminders`.`reminders_id` = `glpi_reminders`.`id`)
LEFT JOIN `glpi_profiles_reminders` ON (`glpi_profiles_reminders`.`reminders_id` = `glpi_reminders`.`id`)
LEFT JOIN `glpi_entities_reminders` ON (`glpi_entities_reminders`.`reminders_id` = `glpi_reminders`.`id`)
WHERE (((`glpi_reminders`.`begin_view_date` IS NULL) OR (`glpi_reminders`.`begin_view_date` < '2022-02-15 10:48:14'))) 
	AND (((`glpi_reminders`.`end_view_date` IS NULL) OR (`glpi_reminders`.`end_view_date` > '2022-02-15 10:48:14'))) 
	AND (`glpi_reminders`.`users_id` = '81' OR `glpi_reminders_users`.`users_id` = '81' OR (`glpi_groups_reminders`.`groups_id` IN ('1') AND (`glpi_groups_reminders`.`entities_id` < '0')) 
	OR (`glpi_profiles_reminders`.`profiles_id` = '4' AND (`glpi_profiles_reminders`.`entities_id` < '0')) 
	OR (`glpi_entities_reminders`.`entities_id` IN ('0', '22', '55', '25', '9', '21', '26', '17', '5', '7', '6', '24', '12', '14', '4', '23', '15', '56', '10', '8', '3', '16', '2', '18', '19', '20', '52', '11', '1', '46', '47', '48', '49', '50', '36', '37', '38', '39', '40', '41', '42', '43', '51', '53', '54', '44', '45', '13', '27', '28', '29', '30', '31', '32', '33', '34', '57'))) 
	AND `glpi_reminders`.`users_id` <> '81'
ORDER BY `name`
```

And I think that the issue is in the where clause:

81 is my user ID, and I think that the issue is in this line: 

`AND (`glpi_reminders`.`users_id` = '81' OR `glpi_reminders_users`.`users_id` = '81' OR (`glpi_groups_reminders`.`groups_id` IN ('1') AND (`glpi_groups_reminders`.`entities_id` < '0')) `

With this query, I can see the public reminders properly:

```
SELECT DISTINCT `glpi_reminders`.*
FROM `glpi_reminders`
	LEFT JOIN `glpi_reminders_users` ON (`glpi_reminders_users`.`reminders_id` = `glpi_reminders`.`id`)
	LEFT JOIN `glpi_groups_reminders` ON (`glpi_groups_reminders`.`reminders_id` = `glpi_reminders`.`id`)
	LEFT JOIN `glpi_profiles_reminders` ON (`glpi_profiles_reminders`.`reminders_id` = `glpi_reminders`.`id`)
	LEFT JOIN `glpi_entities_reminders` ON (`glpi_entities_reminders`.`reminders_id` = `glpi_reminders`.`id`)
WHERE  ((((`glpi_reminders`.`begin_view_date` IS NULL) OR (`glpi_reminders`.`begin_view_date` < '2022-02-15 10:48:14'))) 
	AND (((`glpi_reminders`.`end_view_date` IS NULL) OR (`glpi_reminders`.`end_view_date` > '2022-02-15 10:48:14'))) 
	-- AND (`glpi_reminders`.`users_id` = '81' 
	OR (`glpi_reminders_users`.`users_id` = '81' 
	OR (`glpi_groups_reminders`.`groups_id` IN ('1') 
	AND (`glpi_groups_reminders`.`entities_id` < '0')) 
	OR (`glpi_profiles_reminders`.`profiles_id` = '4' 
	AND (`glpi_profiles_reminders`.`entities_id` < '0')) 
	OR (`glpi_entities_reminders`.`entities_id` IN ('0', '22', '55', '25', '9', '21', '26', '17', '5', '7', '6', '24', '12', '14', '4', '23', '15', '56', '10', '8', '3', '16', '2', '18', '19', '20', '52', '11', '1', '46', '47', '48', '49', '50', '36', '37', '38', '39', '40', '41', '42', '43', '51', '53', '54', '44', '45', '13', '27', '28', '29', '30', '31', '32', '33', '34', '57'))))
	AND `glpi_reminders`.`users_id` <> '81'
ORDER BY `name`
```

### Your GLPI setup information

Operating system: Linux cuess153 5.4.17-2136.302.7.2.el8uek.x86_64 #2 SMP Wed Jan 5 19:12:52 PST 2022 x86_64
PHP 7.4.19 fpm-fcgi (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apcu, bz2, calendar, cgi-fcgi, ctype, curl,
	date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd, openssl,
	pcre, pdo_mysql, pdo_sqlite, session, sockets, sodium, sqlite3, standard, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl,
	zip, zlib)
Setup: max_execution_time="30" memory_limit="128M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="2M" 
Software: Apache/2.4.37 (Oracle Linux) OpenSSL/1.1.1k ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.80 Safari/537.36 Edg/98.0.1108.43
Server Software: MariaDB Server
	Server Version: 10.3.28-MariaDB
	Server SQL Mode: 
	Parameters: root@localhost/glpi_pro
	Host info: Localhost via UNIX socket

**Comment by cconard96** [comment]
I don't see how this would have worked previously. For some reason, there is 11 year old code that restricts the list of public reminders to only ones not created by the current user. Public reminders created by the current user always show up under the private reminders list.

https://github.com/glpi-project/glpi/blob/0fdcecd7e7e8fe77bc780a325d61fc475b4be386/inc/reminder.class.php#L874-L877


I created a reminder under user A and shared it with the root entity recursively. The public reminder showed up under the private reminder list but not the public reminder list. I logged in as another user and the reminder showed up under the public reminder list.
